packages/react-aria-components/src/DropZone.tsx

```diff
@@ -0,0 +1,132 @@
+  /**
+   * Whether the dropzone is currently hovered with a mouse.
+   * @selector [data-hovered]
+   */
+  isHovered: boolean,
+  /**
+   * Whether the dropzone is focused, either via a mouse or keyboard.
+   * @selector :focus
+   */
+  isFocused: boolean,
+  /**
+   * Whether the dropzone is keyboard focused.
+   * @selector [data-focus-visible]
+   */
+  isFocusVisible: boolean,
+  /**
+   * Whether the dropzone is the drop target.
+   * @selector [data-drop-target]
+   */
+  isDropTarget: boolean
+}
+// note: possibly add isDisabled prop in the future
+export interface DropZoneProps extends Omit<DropOptions, 'getDropOperationForPoint' | 'onInsert' | 'onRootDrop' | 'onItemDrop' | 'onReorder'>, RenderProps<DropZoneRenderProps>, SlotProps, AriaLabelingProps {}
+
+export const DropZoneContext = createContext<ContextValue<DropZoneProps, HTMLDivElement>>(null);
+
+function DropZone(props: DropZoneProps, ref: ForwardedRef<HTMLDivElement>) {
+  [props, ref] = useContextProps(props, ref, DropZoneContext);
+  let {dropProps, isDropTarget} = useDrop({...props, ref});
+  let {hoverProps, isHovered} = useHover({});
+  let {focusProps, isFocused, isFocusVisible} = useFocusRing();
+  let [fileTriggerRef, hasFileTrigger] = useSlot(); 
+
+  let textId = useId();
+  let labelProps = useLabels({'aria-labelledby': textId});
+
+  let {clipboardProps} = useClipboard({
+    onPaste: (items) => props.onDrop?.({  
+      type: 'drop',
+      items,
+      x: 0,
+      y: 0,
+      dropOperation: 'copy'
+    })
+  });
+
+  let renderProps = useRenderProps({ 
+    ...props,
+    values: {isHovered, isFocused, isFocusVisible, isDropTarget},
+    defaultClassName: 'react-aria-DropZone'
+  });
+  let DOMProps = filterDOMProps(props);
+  delete DOMProps.id;
+
+  // passed to the FileTrigger component
+  let onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files) {
+      return;
+    } else if (props.onDrop) {
+      props.onDrop({
+        type: 'drop',
+        dropOperation: 'copy',
+        x: 0,
+        y: 0,
+        items: [...e.target.files].map((file) =>
+          ({
+            kind: 'file', 
+            type: file.type, 
+            name: file.name,
+            getFile: () => Promise.resolve(file),
+            getText: () => file.text()
+          })
+        )
+      }
+      );
+    }
+  };
+  
+  return (
+    <Provider
+      values={[
+        [FileTriggerContext, {ref: fileTriggerRef, onChange: onInputChange}],
+        [TextContext, {id: textId, slot: 'heading'}]
+      ]}>
+      <div
+        {...mergeProps(dropProps, hoverProps, DOMProps, clipboardProps)} 
+        {...renderProps}
+        ref={ref}
+        slot={props.slot}
+        tabIndex={-1}
+        data-hovered={isHovered || undefined}
+        data-focused={isFocused || undefined}
+        data-focus-visible={isFocusVisible || undefined}
+        data-drop-target={isDropTarget || undefined} >
+        {!hasFileTrigger &&
```

Comment:
I think the button element needs to be non-conditional, and the `dropProps` should go on it rather than the outer div. Right now, this div does not have a role, so when dropping with VoiceOver there the announcement doesn't say it's a button. Also clipboardProps is duplicated right now and should probably only go on the button.